src/test/scala/mesosphere/mesos/ResourceMatcherTest.scala

```diff
@@ -1179,6 +1180,46 @@ class ResourceMatcherTest extends UnitTest with Inside {
       resourceMatchResponse shouldBe a[ResourceMatchResponse.NoMatch]
       resourceMatchResponse.asInstanceOf[ResourceMatchResponse.NoMatch].reasons.head shouldEqual DeclinedScarceResources
     }
+
+    "correctly match offers in case of app specific override and no Persistent Volume involved" in {
+
+      val overrideCases = Table(
+        ("gpu_scheduling_behavior", "GPU_SCHEDULING_BEHAVIOR", "expected"),
+        ("unrestricted", Some("restricted"), "NoMatch"),
+        ("unrestricted", None, "Match"),
+        ("restricted", Some("unrestricted"), "Match"),
+        ("restricted", None, "NoMatch")
+      )
+
+      forAll(overrideCases) { (gpuSchedulingBehavior, overrideLabel, expected) =>
+        val gpuConfig = AllConf.withTestConfig(
+          "--draining_seconds", "300",
+          "--gpu_scheduling_behavior", gpuSchedulingBehavior,
+          "--enable_features", "gpu_resources")
+        val offer = MarathonTestHelper.makeBasicOffer(gpus = 4)
+          .build()
+        val app = AppDefinition(
+          id = "/test".toRootPath,
+          resources = Resources(cpus = 1.0, mem = 128.0, disk = 0.0),
+          portDefinitions = PortDefinitions(0, 0),
+
+          labels = overrideLabel.map(label => Map("GPU_SCHEDULING_BEHAVIOR" -> label)).getOrElse(Map.empty)
```

Comment:
Nit: Is `overrideLabel.map( ... Map)` not returning a `Option[Map]`?

Reply:
it does!

Reply:
Oh, I missed the `getOeElse` part.